src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -76,22 +108,12 @@ internal void GrabDiagnostics(DiagnosticBag addTo)
             if (!diags.IsDefault)
             {
                 addTo.AddRange(diags);
-            }
-        }
-
-        private void AddDiagnostics(ImmutableArray<Diagnostic> diagnostics)
-        {
-            // Atomic update operation. Applies a function (Concat) to a variable repeatedly, until it "gets through" (isn't in a race condition with another concat)
-            var oldDiags = _diagnostics;
-            while (true)
-            {
-                var newDiags = oldDiags.IsDefault ? diagnostics : oldDiags.Concat(diagnostics);
-                var overwriteDiags = ImmutableInterlocked.InterlockedCompareExchange(ref _diagnostics, newDiags, oldDiags);
-                if (overwriteDiags == oldDiags)
+                addTo.AddRange(_lazyParametersAndDiagnostics.Diagnostics);
+                if (_lazyTypeParameterConstraintsAndDiagnostics != null)
```

Comment:
`_lazyTypeParameterConstraintsAndDiagnostics` is only computed if there are constraints. Added a comment to clarify.